functions/gh_repo/workload.py

```diff
@@ -4,17 +4,23 @@
 import time
 
 def handle(req, syscall):
-    key = "github/%s/%s.tgz" % (req["repository"]["full_name"], req["after"])
-    meta_key = "github/%s/_meta" % (req["repository"]["full_name"])
-    workflow_key = "github/%s/_workflow" % (req["repository"]["full_name"])
+#    key = "github/%s/%s.tgz" % (req["repository"]["full_name"], req["after"])
+#    meta_key = "github/%s/_meta" % (req["repository"]["full_name"])
+#    workflow_key = "github/%s/_workflow" % (req["repository"]["full_name"])
+    usergroup = req['repository']['full_name']
+    key = '/gh_repo/%s/%s.tgz' % (req['repository']['full_name'], req['after'])
+    base_dir = '/gh_repo/%s' % usergroup
+    meta_path = '/start_assignment/%s/_meta' % usergroup
+    workflow_path = '/start_assignment/%s/_workflow' % usergroup
 
-    metadataString = syscall.read_key(bytes(meta_key, "utf-8")) or "{}"
+    metadataString = syscall.fsread(meta_path) or "{}"
     if metadataString:
         metadata = json.loads(metadataString)
-        workflow = json.loads(syscall.read_key(bytes(workflow_key, "utf-8")) or "[]")
+        workflow = json.loads(syscall.fsread(workflow_path) or "[]")
 
         resp = syscall.github_rest_get("/repos/%s/tarball/%s" % (req["repository"]["full_name"], req["after"]));
-        syscall.write_key(bytes(key, "utf-8"), resp.data)
+        syscall.fscreate_file(base_dir, '%s.tgz' % req['after'], syscall.get_current_label())
```

Comment:
Ideally, here we would label the output file with secrecy corresponding to the owners of the repository.